python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -455,25 +468,14 @@ class RandomForestRegressor(Base):
 
     def _predict_model_on_gpu(self, X, algo,
                               convert_dtype, task_category=1):
-
-        cdef ModelHandle cuml_model_ptr
         X_m, _, n_rows, n_cols, _ = \
             input_to_dev_array(X, order='C', check_dtype=self.dtype,
                                convert_to_dtype=(self.dtype if convert_dtype
                                                  else None),
                                check_cols=self.n_cols)
 
-        cdef RandomForestMetaData[float, float] *rf_forest = \
-            <RandomForestMetaData[float, float]*><size_t> self.rf_forest
-
         task_category = 1  # for regression
-        build_treelite_forest(& cuml_model_ptr,
-                              rf_forest,
-                              <int> n_cols,
-                              <int> task_category,
-                              <vector[unsigned char] &> self.model_pbuf_bytes)
-        mod_ptr = <size_t> cuml_model_ptr
-        treelite_handle = ctypes.c_void_p(mod_ptr).value
+        treelite_handle = self._convert_to_treelite(task_category)
```

Comment:
It would also be nice to have a method that converts a cuML RF model into a FIL model. This method can be placed in RandomForestClassifier/Regressor, ForestInference or both.